src/pushsource/_impl/backend/koji_source.py

```diff
@@ -509,6 +519,104 @@ def _push_items_from_container_build(self, nvr, meta):
 
         return out
 
+    def _push_items_from_vmi_build(self, nvr, meta):
+        LOG.debug("Looking for virtual machine images on %s, %s", nvr, meta)
+
+        if not meta:
+            message = "Virtual machine image build not found in koji: %s" % nvr
+            LOG.debug(message)
+            raise ValueError(message)
+
+        if meta.get("container_koji_task_id"):
+            message = "The build %s is for container images." % nvr
+            LOG.debug(message)
+            raise ValueError(message)
+
+        # VMI has the {'typeinfo': {'image':{}}} on extra
+        extra = meta.get("extra") or {}
+        typeinfo = extra.get("typeinfo") or {}
+        image = typeinfo.get("image")
+
+        if image is None:
+            message = "Build %s not recognized as a virtual machine image build" % nvr
+            LOG.debug(message)
+            raise ValueError(message)
+
+        build_id = meta["id"]
+
+        archives = self._get_archives(build_id)
+
+        # Supported types of VMI archives
+        ami_types = ["raw", "raw-xz"]
+        azure_types = ["vhd", "vhd-compressed"]
+        cloud_types = {}
+
+        for k in ami_types:
+            cloud_types.setdefault(k, AmiPushItem)
+
+        for k in azure_types:
+            cloud_types.setdefault(k, VHDPushItem)
+
+        # Collect all known VM image archives
+        vmi_archives = [
+            elem
+            for elem in archives
+            if elem.get("btype") == "image"
+            and elem.get("type_name") in cloud_types.keys()
+        ]
+
+        out = []
+
+        # Generate the respective PushItem type for each archive
+        for archive in vmi_archives:
+            path = self._pathinfo.typedir(meta, archive["btype"])
+            item_src = os.path.join(path, archive["filename"])
+
+            extra = archive.get("extra") or {}
+            image = extra.get("image") or {}
+            arch = image.get("arch")
+
+            klass = cloud_types.get(archive["type_name"]) or VMIPushItem
+
+            rel_klass = klass._RELEASE_TYPE
+
+            # The given name will be composed by the product name and cloud
+            # marketpalce name, like "rhel-azure" or "rhel-sap-azure".
+            # We just want to get the product name without the cloud name,
+            # like "rhel" and "rhel-sap" in our examples respectively.
+            splitted_name = meta["name"].split("-")
+            if len(splitted_name) > 1:
+                product = "-".join(splitted_name[:-1])
+            else:
+                product = splitted_name[0]
+
+            release = rel_klass(
+                arch=arch,
+                date=(meta.get("completion_time") or "").split(" ")[0],
+                product=product.upper(),
+                version=meta["version"],
+                respin=int(meta["release"]),
```

Comment:
Are you sure that `meta["release"]` is always an integer?

Did anyone provide a link to an example build?  I'm looking at https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=2350393 and release is not an integer there, which causes this to crash:

```
$ pushsource-ls 'koji:https://brewhub.engineering.redhat.com/brewhub/?vmi_build=rhel-azure-7.9-20230110.sp.22'
[...]
  File "/home/rmcgover/src/pushsource/src/pushsource/_impl/backend/koji_source.py", line 598, in _push_items_from_vmi_build
    respin=int(meta["release"]),
           ^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '20230110.sp.22'
```

The code should be usable with that build, right? (or can you provide at least one real build it should work against?)

Reply:
You're right, this is a valid `release`.

I'll work on fixing this, thanks a lot!

Reply:
I've changed it to load the integer from just the first part of the dots if it exists, like `20230201.sp.22` would result into `20230201`.

If it's not possible to convert it to `int` it will be `0` so the program shouldn't fail again.

I also added this string into the tests to cover it, please let me know wdyt.